get-started/part2.md

```diff
@@ -8,125 +8,129 @@ description: Learn how to write, build, and run a simple app -- the Docker way.
-stack, defining the interactions of all the services, covered in
-[Part 5](part5.md).
+This tutorial teaches you how to build a simple Python web application (with
+[Flask](http://flask.pocoo.org/){:target="_blank"}) that prints **"Hello, Moby Dock!"**
+on a blue canvas. We later add a [Redis](https://hub.docker.com/_/redis/){:target="_blank"}
+database to count the number of visitors to our site.
+
+Here in part 2, we start by building a Docker application image, `hellomoby`,
+then run it as a single container on localhost. We also tag and push the new
+image to Docker's public registry via [Docker Hub](https://hub.docker.com/){:target="_blank"}.
+
+In [Part 3](part3.md), we run our image as a "Docker Service" so we can scale
+the number of containers and do other operations. In [Part 4](part4.md), we
+employ container orchestration and distribute the multiple containers running
+our application across multiple nodes. In [Part 5](part5.md), we add a service
+(for a redis database) and rebuild our image as a Docker stack.
 
 - Stack
-- Services
+- Docker Service
 - **Container** (you are here)
 
-## Your new development environment
-
-In the past, if you were to start writing a Python app, your first
-order of business was to install a Python runtime onto your machine. But,
-that creates a situation where the environment on your machine has to be just
-so in order for your app to run as expected; ditto for the server that runs
-your app.
+## Define app image with Dockerfile
 
-With Docker, you can just grab a portable Python runtime as an image, no
-installation necessary. Then, your build can include the base Python image
-right alongside your app code, ensuring that your app, its dependencies, and the
-runtime, all travel together.
+Typically, creating a Python application requires that you install a Python
+runtime, along with the server, and configure it all _perfectly_ on your
+platform for the app to run as expected. With Docker, you simply pull a portable
+Python runtime image from [Docker Hub](https://hub.docker.com/_/python/){:target="_blank"}
+(or [Docker Store](https://store.docker.com/images/python)) and build it with
+your application code (as another image).
 
-These portable images are defined by something called a `Dockerfile`.
+To ensure the integrity and portability of an application, we build images with
+a [Dockerfile](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/).
+A `Dockerfile` defines the environment inside a container. Access to resources,
+such as networking interfaces and disk drives, is virtualized inside this
+environment. The container is isolated from the rest of your system, so you must
+map ports to the outside world, and be specific about what files you want to
+copy into that environment. Let's start.
 
-## Define a container with `Dockerfile`
+Create an empty directory and change directories into this new directory:
```

Comment:
Do you want to make these numbered steps?

Reply:
I know ... I've been following the existing format but this could be a good change.